crates/wasm-mutate-stats/src/bin/wasm-mutate-stats.rs

```diff
@@ -0,0 +1,327 @@
+                                                report.add_wasm_hash(hash);
+
+                                                for (idx, mhash) in mhashes.iter().enumerate() {
+                                                    report.add_obj_hash(mhash.clone(), idx);
+                                                }
+
+                                                if report.rotations % 100 == 0 {
+                                                    // Save report
+                                                    self.save_report(&report, &name);
+                                                    if self.print {
+                                                        report.print();
+                                                    }
+                                                }
+                                            }
+                                            Err(e) => report
+                                                .errors
+                                                .push((report.rotations as u64, format!("{}", e))),
+                                        }
+                                    }
+                                    // All generated Wasm should be valid
+                                    Err(e) => panic!("{}", e),
+                                }
+                            }
+                            Err(e) => report
+                                .errors
+                                .push((report.rotations as u64, format!("{}", e))),
+                        }
+                    }
+                }
+                None => {
+                    break;
+                }
+            }
+        }
+    }
+
+    fn hash(&self, data: Vec<u8>) -> String {
+        // By default md5 is the hashes here
+        format!("{:x}", Md5::digest(&data))
+    }
+
+    /// Compile a module using `mutated` as the input Wasm
+    ///
+    fn compile(
+        &self,
+        mutated: &Vec<u8>,
+        modulename: &String,
+        rotation: usize,
+    ) -> Result<Vec<String>, wasm_mutate_stats::Error> {
+        // This is suuuper ugly
+        let mut buffer = fs::File::create(format!("{}.mutated", modulename))?;
+        buffer.write(&mutated)?;
+
+        let mut hashes = Vec::new();
+        let wasmtime_bin = std::env::var("WASMTIME").expect("WASMTIME variable is not set");
+        // Execute each compiler configuration
+        self.compiler_configs
+            .iter()
+            .map(|config| {
+                let name = format!("{}.{}.{}", modulename, rotation, config.join(""));
+
+                let command = Command::new(&wasmtime_bin)
+                    .arg("compile")
+                    .arg(format!("{}.mutated", modulename))
+                    .args(config)
+                    .arg("-o")
+                    .arg(format!("{}.obj", name))
+                    .spawn()
+                    .expect("Process cannot be launched");
```

Comment:
Rather than spawning a `wasmtime` subprocess, which requires configuring which wasmtime CLI to use and env vars and all that, we can use the `wasmtime` API to do it in this process.

```rust
// create the engine outside this function and pass it in as an argument, so that
// it is shared across all modules.
let engine = wasmtime::Engine::default();

// ...

let module = wasmtime::Module::new(&engine, &mutated)
    .expect("the mutated Wasm should be valid");
let obj = module.serialize()?
```

This way, we wouldn't need to write the mutated Wasm out to a file either.